test/test_data.py

```diff
@@ -5,91 +5,388 @@
+    assert streams[i]["info"]["nominal_srate"][0] == "10"
+    assert streams[i]["info"]["channel_format"][0] == "string"
+    assert streams[i]["info"]["created_at"][0] == "50942.723319709003"
+    assert streams[i]["info"]["desc"][0] is None
+    assert streams[i]["info"]["uid"][0] == "xdfwriter_11_str"
 
-        clock_times = np.asarray([])
-        clock_values = np.asarray([])
+    # Info added by pyxdf
+    assert streams[i]["info"]["stream_id"] == 0x02C0FFEE
+    assert streams[i]["info"]["effective_srate"] == pytest.approx(10)
+    assert streams[i]["info"]["segments"] == [(0, 8)]
 
-        np.testing.assert_array_equal(streams[1]["clock_times"], clock_times)
-        np.testing.assert_array_almost_equal(streams[1]["clock_values"], clock_values)
+    # Footer
+    assert streams[i]["footer"]["info"]["first_timestamp"][0] == "5.1"
+    assert streams[i]["footer"]["info"]["last_timestamp"][0] == "5.9"
+    assert streams[i]["footer"]["info"]["sample_count"][0] == "9"
+    first_clock_offset = streams[i]["footer"]["info"]["clock_offsets"][0]["offset"][0]
+    assert first_clock_offset["time"][0] == "50979.76"
+    assert first_clock_offset["value"][0] == "-.01"
 
-        streams, header = load_xdf(
-            file, jitter_break_threshold_seconds=0.001, jitter_break_threshold_samples=1
-        )
+    # Time-series data
+    s = [
+        [
+            '<?xml version="1.0"?><info><writer>LabRecorder xdfwriter'
+            "</writer><first_timestamp>5.1</first_timestamp><last_timestamp>"
+            "5.9</last_timestamp><sample_count>9</sample_count>"
+            "<clock_offsets><offset><time>50979.76</time><value>-.01</value>"
+            "</offset><offset><time>50979.86</time><value>-.02</value>"
+            "</offset></clock_offsets></info>"
+        ],
+        ["Hello"],
+        ["World"],
+        ["from"],
+        ["LSL"],
+        ["Hello"],
+        ["World"],
+        ["from"],
+        ["LSL"],
+    ]
+    t = np.array([5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9])
+    assert streams[i]["time_series"] == s
+    np.testing.assert_allclose(streams[i]["time_stamps"], t)
+
+    # Clock offsets - does not match the footer but it is in the file
+    clock_times = np.asarray([])
+    clock_values = np.asarray([])
+
+    np.testing.assert_equal(streams[i]["clock_times"], clock_times)
+    np.testing.assert_equal(streams[i]["clock_values"], clock_values)
+
+    streams, header = load_xdf(
+        path,
+        synchronize_clocks=synchronize_clocks,
+        dejitter_timestamps=dejitter_timestamps,
+        jitter_break_threshold_seconds=0.001,
+        jitter_break_threshold_samples=1,
+    )
+    if dejitter_timestamps:
         assert streams[0]["info"]["segments"] == [(0, 0), (1, 3), (4, 8)]
+    else:
+        assert streams[0]["info"]["segments"] == [(0, 8)]
+
+
+@pytest.mark.parametrize("dejitter_timestamps", [False, True])
+@pytest.mark.parametrize("synchronize_clocks", [False, True])
```

Comment:
It seems like you do not use these two parameters in your test. Depending on if this was intentional, I suggest removing these parameterizations or adding tests for these settings.

Reply:
These parameters are used, they are passed to `load_xdf`.

```
test/test_data.py::test_minimal_file[False-False] PASSED           [ 12%]
test/test_data.py::test_minimal_file[False-True] PASSED            [ 25%]
test/test_data.py::test_minimal_file[True-False] PASSED            [ 37%]
test/test_data.py::test_minimal_file[True-True] PASSED             [ 50%]
test/test_data.py::test_empty_streams_file[False-False] PASSED     [ 62%]
test/test_data.py::test_empty_streams_file[False-True] PASSED      [ 75%]
test/test_data.py::test_empty_streams_file[True-False] PASSED      [ 87%]
test/test_data.py::test_empty_streams_file[True-True] PASSED       [100%]
```

Thanks for adding the missing `skipif`!

Reply:
What I meant is that in `test_empty_streams`, you are not testing for the different values of `dejitter_timestamps` and `synchronize_clocks`. All tests are completely independent of whether these are set to `True` or `False`.

Reply:
> All tests are completely independent of whether these are set to True or False.

The test is now conditional on synchronize_clocks.

I've kept the parametrization of dejitter even though it has minimal effect on idealised time-stamps - that feels like a reassuring thing to validate to me, but also happy to remove it if you prefer.